File: Python Code/Daily_Flash_Code_0/Week 5/10 Feb/24-DailyFlash_Solutions/24-DailyFlash_Solutions/10_Feb_Solutions_Three/Python/prog2.py
from array import *

arr = input("Enter octal num : ")
b = array('i', [])
for i in range(3):
    b.append(0)
l = len(arr);
for i in range(l):
    n = ord(arr[i]) - 48
    k = 0
    b[2] = 0
    b[1] = 0
    b[0] = 0
    while(n != 0):
            rem = n % 2
            b[k] = rem
            k += 1
            n = int(n / 2)
    print(b[2],b[1],b[0], end = " ");	

File: Python Code/Daily_Flash_Code_0/Week 5/10 Feb/24-DailyFlash_Solutions/24-DailyFlash_Solutions/10_Feb_Solutions_Three/C/prog2.c
#include<stdio.h>
#include<string.h>
int main(void) {
	char arr[10];
	int bin[3];
	printf("Enter octal num : ");
	fgets(arr, sizeof(arr), stdin);
	if(arr[strlen(arr) - 1] == '\n') {
		arr[strlen(arr) - 1] = '\0';
	}
	int n, k, rem;
	int len = strlen(arr);
	for(int i = 0; i < len; i++) {
		n = arr[i] - 48;
		k = 0;
		bin[2] = 0;
		bin[1] = 0;
		bin[0] = 0;
		while(n != 0) {
			rem = n % 2;
			bin[k] = rem;
			k++;
			n = n / 2;
		}
		printf("%d%d%d", bin[2], bin[1], bin[0]);	
	}
}
